scanpy/preprocessing/_cell_hashing.py

```diff
@@ -0,0 +1,109 @@
+from typing import Iterable
+
+from anndata import AnnData
+
+from scipy.signal import argrelextrema
+from scipy.stats import gaussian_kde
+import numpy as np
+import itertools
+import pandas as pd
+
+
+def _demultiplex_per_barcode(x):
+    """
+    Approximates method in https://www.nature.com/articles/s41592-019-0433-8
+    MULTI-seq: sample multiplexing for single-cell RNA
+    sequencing using lipid-tagged indices
+    """
+    percentiles = np.percentile(x, q=np.arange(.01, 99.9, step=.1))
+    kde = gaussian_kde(x)
+    dist = kde(percentiles)
+    maxima = argrelextrema(dist, np.greater_equal)[0]
+    min_maxima = np.min(maxima)
+    max_maxima = np.max(maxima)
+    intermediate_thresh = (percentiles[min_maxima] + percentiles[max_maxima]) / 2
+    all_low_maxima = maxima[np.where(percentiles[maxima] < intermediate_thresh)[0]]
+    low_maxima = all_low_maxima[np.argmax(dist[all_low_maxima])]
+    info = []
+    for q in range(1, 100):
+        threshold = np.percentile([percentiles[low_maxima],
+                                   percentiles[max_maxima]], q=q)
+        info.append([x >= threshold, threshold])
+    return info, [percentiles, dist]
+
+
+def demultiplex_hashing(adata: AnnData,
+                        percentile_candidates: Iterable[int] = None,
+                        inplace: bool = True):
+    """
+    Demultiplexing cell hashing data
+
+    Params
+    ------
+    adata
+        The annotated data matrix of shape ``n_obs`` × ``n_vars``. Rows correspond
+        to cells and columns to genes.
+    percentile_candidates
```

Comment:
Can you add if the adata object needs any particular formatting or pre-processing steps? Maybe here or in the description. For example for 10x, just processing the sample with cell ranger is enough or other steps are required?